spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java

```diff
@@ -878,10 +874,6 @@ private synchronized void clearState(final File fileToWriteTo, final FileState s
 	}
 
 	private static boolean rename(File source, File target) throws IOException {
-		return (nioFilesPresent && filesMove(source, target)) || source.renameTo(target);
-	}
-
-	private static boolean filesMove(File source, File target) throws IOException {
 		Files.move(source.toPath(), target.toPath(), StandardCopyOption.REPLACE_EXISTING);
 		return true;
```

Comment:
This can return void. We can now remove the `if` from `if(rename...` and the log about falling back to copy/delete in `handleFileMessage()`.

Also, the logic in `renameTo()` that deletes an existing file can be removed and the if tests and associated `throw` s can be removed also, since `Files.move` will throw an exception.